wire/protobuf/serializer.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2022 - See NOTICE file for copyright holders.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package protobuf
+
+import (
+	"encoding/binary"
+	"io"
+	"time"
+
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/proto"
+	"perun.network/go-perun/wire"
+)
+
+// Serializer implements methods for encoding/decoding
+// envelopes using protobuf encoding.
+type Serializer struct{}
+
+// func init() {
+// 	wire.SetEnvelopeSerializer(Serializer{})
+// }
```

Comment:
Hm?

Reply:
The PR is not yet complete. I will adding a few more commits, that will eventually remove this.